static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+    const periods_all = xx.map((x) => Math.log10(1.0 / x)).reverse();
+    const power_all = yy.reverse();
+
+    // select only the range we want (< periodmax)
+    const goodi = periods_all
+      .map((e, i) => (e < kwargs.periodmax ? i : undefined))
+      .filter((x) => x);
+    const periods = goodi.map((i) => periods_all[i]);
+    const power = goodi.map((i) => power_all[i]);
+
+    const g = new Dygraph(glsplotRef.current, transpose([periods, power]), {
+      clickCallback(e) {
+        const x = e.offsetX;
+        const y = e.offsetY;
+        const dataXY = g.toDataCoords(x, y);
+        setBestp(10 ** dataXY[0]);
+      },
+      axes: {
+        x: { logscale: false },
+        y: {},
+      },
+      strokeWidth: 2,
+      xlabel: "log Period [day]",
+      ylabel: "Power",
+      labels: ["log Period [day]", "Power"],
+    });
+
+    plotline(g, kwargs.periodbest);
+  }
+
+  useEffect(() => {
+    if (!photometry) {
+      dispatch(photometryActions.fetchSourcePhotometry(id));
+    }
+    if (photometry) {
+      setInstruments([...new Set(photometry.map((x) => x.instrument_name))]);
+      setFilters([...new Set(photometry.map((x) => x.filter))]);
+    }
+
+    if (photometry && !run) {
+      const data = photometry.filter(
+        (x) =>
+          x.filter === params.filter && x.instrument_name === params.instrument
+      );
+      const times = data.map((x) => x.mjd);
+      const mag = data.map((x) => x.mag);
+      const magerr = data.map((x) => x.magerr);
+      if (!plotted) {
+        plotdata(times, mag, magerr);
+        setPlotted(true);
+      }
+
+      // filtered times
+      const ftimes = times.filter((x) => x >= mjdmin && x <= mjdmax);
+      const fmag = mag.filter(
+        (x, i) => times[i] >= mjdmin && times[i] <= mjdmax
+      );
+      const fmagerr = magerr.filter(
+        (x, i) => times[i] >= mjdmin && times[i] <= mjdmax
+      );
+
+      const gls = GLS(ftimes, fmag, {
+        e_y: fmagerr,
+        ofac: parseInt(params.ofac, 10),
+        fbeg: null,
+        fend: null,
+        ls: null,
+      });
+      plotGLS(gls.f, gls.p, {
+        periodbest: 1 / gls.fbest,
+        periodmax: gls.tbase / 3.5,
+      });
+      setBestp(1 / gls.fbest);
+      setRun(true);
+    }
+    if (run) {
+      const data = photometry.filter(
+        (x) =>
+          x.filter === params.filter &&
+          x.instrument_name === params.instrument &&
+          x.mjd >= mjdmin &&
+          x.mjd <= mjdmax
+      );
+      const times = data.map((x) => x.mjd);
+      const mag = data.map((x) => x.mag);
+      const title = `P=${bestp?.toFixed(8)} d (${params.instrument}: ${
+        params.filter
+      })`;
+      plotphased(times, mag, bestp, title);
+    }
+  }, [id, photometry, bestp, run, dispatch]);
+
+  const componentRef = useRef();
+
+  const initialFormState = {
+    ...params,
+  };
+
+  const onSubmit = () => {
+    const formData = {
+      ...initialFormState,
+      ...getValues(),
```

Comment:
the form values will be provided to `onSubmit` as an argument upon form submission, so getValues shouldn't be needed here

Reply:
Done.